### PR TITLE
Added some memory addresses

### PR DESCRIPTION
**Pull request by TheMachineAmbassador**

### CODE REVIEW

File: SH2/SH2 PC memory addresses [hunk 1]
```diff
@@ -251,9 +254,12 @@ sh2pc.exe+19BBF8A (01DBBF8A) byte - Number of saves
 sh2pc.exe+19BBF94 (01DBBF94) float - In-game timer (IGT) ...aka total time in current playthrough
 sh2pc.exe+19BBF98 (01DBBF98) float - Walking distance
 sh2pc.exe+19BBF9C (01DBBF9C) float - Running distance
-sh2pc.exe+19BBF8E (01DBBF8E) 2 bytes - Number of items collected (not sure if it counts bonus secret items)
+sh2pc.exe+19BBF8E (01DBBF8E) 2 bytes - Number of items collected
+sh2pc.exe+19BBF88 (01DBBF88) byte - Number of secret items collected
 sh2pc.exe+19BBF90 (01DBBF90) 2 bytes - Shooting kills
 sh2pc.exe+19BBF92 (01DBBF92) 2 bytes - Melee kills
 sh2pc.exe+19BBFA0 (01DBBFA0) float - Boat stage time
 sh2pc.exe+19BBFA0 (01DBBFA4) float - Boat max speed
```
Comment (TheMachineAmbassador) [inline SH2/SH2 PC memory addresses:262]: This line is wrong i forgot the change it. Boat max speed should be sh2pc.exe+19BBFA4

**Comment by TheMachineAmbassador** [comment]
RNG Value For Encrypting Save Data!!!
In save system in the game this rng value only used for creating new save and writing on to it. After new save file created game hides the encrypiton hash byte in the save file and XOR's save data with that byte. But the thing is you cant use the hash to re-encrypt save data. sh2pcsaveXX.dat files are used for holding player data and which level you're currently on etc... sh2pc.sys hash own encryption and decryption methods aswell. Like this guy posted on issue for enchanced edition https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/47#issuecomment-803444691 the decrypted sh2pc.sys looks like that. You can see data values decrypted because after you save the file game turns it back to the decryoted data value again. Now what is sh2pc.sys doing? It is orders save values for listing, holds the game difficulty modes (but i think these modes are just text. They doesen't affects the game difficulty the real game difficulty are in the sh2pcsaveXX.dat file), which game mode you're playing (regular or BFaW) and 8 byte flag value for stored in save data file for the correct that file is save data. I didn't test but there is may be bug on 2tb fix because the sh2pc.sys file only holds 3kb for the listing and 1 save is costs 24 bytes so theoretically the game only holds 128 save data (again i didnt test is this correct after i test it out i will post here). So is game doesen't extend this ? No it doesen't because the file size is fixed when created. I actually tried to make a save editor and the easiest part was decrypting sh2pcsaveXX.dat because i didnt need to deal with RNG. But when its come to part of re-encrypting data its all messed up. Even i encrypt the data i had to decrypt and encrypt sh2pc.sys file too. It was too messy (sh2pc.sys uses RNG again) and i had no time to do all process for sh2pc.sys file too. So Good Luck Everyone!!! I just wanted to give some info about how save system works in the game. For now i will try to fix this issue https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/47 and then my works in this game will be completed i guess :) when i have time i'm gonna look at the save editor again!

**Comment by TheMachineAmbassador** [comment]
I tested how many save can handle sh2pc.sys file. ss Result is only 100 after whole save encryption completed game writes save data to file with this function 0x456870 i tried to change size of written data but it didnt work. If you want to understand what sh2pc.sys file is you can real time test it from this byte 0x936b68 when you create new save or overwrite a save data first data writes to that memory location then it copies to 0x933608 this location and then game sends data to encryption function and finally it sends to function that i share above.

**Comment by Polymega** [comment]
> For now i will try to fix this issue https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/47 and then my works in this game will be completed i guess :) when i have time i'm gonna look at the save editor again!

Thanks for taking an interest in this! I believe we have this issue resolved now, but have yet to release a public update for the project that includes our fix for this. [You can find a test build with this fix here.](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/47#issuecomment-809052056)

Here is the commit for this fix:
https://github.com/elishacloud/Silent-Hill-2-Enhancements/commit/2cbc3bf1cdb4cc7fb570634a4551781d4faef5bf

And here is an addendum to this fix that stops previous game sounds once the Game Result screen is loaded in:
https://github.com/elishacloud/Silent-Hill-2-Enhancements/commit/2ef6805d5013be6a2f633d00e5fd021bc100cbb2

-------------------------------

But if you'd still like to help us with the project--and for something related to save files--[we still need help with this feature, if you're interested?](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/209) 

I can't remember where we discussed it, but _I think_ we've determined that it's most likely impossible to preserve the "game load" SFX when loading a save file while in-game (as the game stops/resets all audio upon file load). 

However, we'd still really like a way to play the "game save" SFX when selecting **Save Game** through the PC pause menu (the same as you'd hear when interacting with a red square).

**Comment by TheMachineAmbassador** [comment]
> > For now i will try to fix this issue [elishacloud/Silent-Hill-2-Enhancements#47](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/47) and then my works in this game will be completed i guess :) when i have time i'm gonna look at the save editor again!
> 
> Thanks for taking an interest in this! I believe we have this issue resolved now, but have yet to release a public update for the project that includes our fix for this. [You can find a test build with this fix here.](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/47#issuecomment-809052056)
> 
> Here is the commit for this fix: [elishacloud/Silent-Hill-2-Enhancements@2cbc3bf](https://github.com/elishacloud/Silent-Hill-2-Enhancements/commit/2cbc3bf1cdb4cc7fb570634a4551781d4faef5bf)
> 
> And here is an addendum commit for this fix that stops previous game sounds once the Game Result screen is loaded in: [elishacloud/Silent-Hill-2-Enhancements@2ef6805](https://github.com/elishacloud/Silent-Hill-2-Enhancements/commit/2ef6805d5013be6a2f633d00e5fd021bc100cbb2)
> 
> But if you'd still like to help us with the project--and for something related to save files--[we still need help with this feature, if you're interested?](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/209)
> 
> I can't remember where we discussed it, but _I think_ we've determined that it's most likely impossible to preserve the "game load" SFX when loading a save file while in-game (as the game stops/resets all audio upon file load).
> 
> However, we'd still really like a way to play the "game save" SFX when selecting **Save Game** through the PC pause menu (the same as you'd hear when interacting with a red square).

Of course ! i can look at that issue. You mean this sound right?
[save.zip](https://github.com/JokieW/SilentHillDatabase/files/8127924/save.zip)
e

**Comment by Polymega** [comment]
Yup, that's the sound! I'll ping you over on the ticket for this. Thanks again!

**Pull request merged**